File: Submissions/#0087_div2/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();

    int answer = 0;
    int current = 0;

    for (int i = 0; i < n; i++) {
      int m = sc.nextInt();
      int p = sc.nextInt();
      current -= m;
      current += p;
      answer = Math.max(answer, current);
    }

    sc.close();

    System.out.println(answer);

  }

}

File: Submissions/#0087_div2/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int n;
  cin >> n;

  int answer = 0;
  int current = 0;
  
  for (int i = 0; i < n; i++) {
    int m, p;
    cin >> m >> p;
    current -= m;
    current += p;
    answer = max(answer, current);
  }

  cout << answer << endl;

}